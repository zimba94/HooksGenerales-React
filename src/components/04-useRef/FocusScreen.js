import React, {useRef} from 'react'

export const FocusScreen = () => {

    const inputRef = useRef();
    const handleClick = () => {
        // document.querySelector('input').select();
        inputRef.current.select();
    }
    return (
        <div>
            <h1>Focus Screen</h1>
            <hr />
            <input 
                className="form-control"
                placeholder="Su nombre"
                ref={inputRef}
            />

            <button 
                className="btn btn-outline-primary mt-5"
                onClick = {handleClick}
            >
                Focus
            </button>

        </div>
    )
}
