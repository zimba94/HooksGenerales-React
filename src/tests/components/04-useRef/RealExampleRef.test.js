import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";

describe('Pruebas en el componente RealExampleRef', () => {

    const wrapper = shallow(< RealExampleRef />);

    test('Debe mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    });

    test('Debe de mostrar el componente </MultipleCustomHooks>', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    });
    
    
});
