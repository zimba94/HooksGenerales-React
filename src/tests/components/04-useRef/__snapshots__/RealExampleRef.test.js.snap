// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente RealExampleRef Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
